crates/core/src/db/datastore/traits.rs

```diff
@@ -38,15 +38,22 @@ pub struct TxData {
 }
 
 pub trait Data: Into<ProductValue> {
-    fn view(&self) -> &ProductValue;
+    fn view(&self) -> Cow<'_, ProductValue>;
 }
 
 pub trait DataRow: Send + Sync {
     type RowId: Copy;
 
     type DataRef<'a>;
 
-    fn view_product_value<'a>(&self, data_ref: Self::DataRef<'a>) -> &'a ProductValue;
+    /// Return a `Cow`, which currently will always be `Borrowed`,
```

Comment:
Is this relevant for the user or implementor of the trait? Is there anything I’d need to do differently depending on whether the cow is borrowed or owned? How does borrowing hide any facts about the stored format?

Reply:
The core interface of `Cow` is that it implements `Deref`. `Deref` gives a ref with a lifetime bounded by the `Cow`, not by (in our case) the TX. This is in contrast to our current `view_product_value`, which gives a ref with a lifetime bounded by the TX, which may outlive the `DataRef`. The punchline here is that nowhere in our code do we depend on that longer lifetime, so `Cow` (with deref coercion) is a drop-in change.

Returning `Cow` here hides the stored format in the sense that the current datastore can return `Cow::Borrowed`, whereas the new datastore will return `Cow::Owned`, and consumers (at least in our codebase as it exists now) are blissfully ignorant of the difference.

Reply:
Well, I was asking to clarify this in the commentary.